build/lib/release.sh

```diff
@@ -386,8 +386,8 @@ EOF
         # If we are building an official/alpha/beta release we want to keep
         # docker images and tag them appropriately.
         local release_docker_image_tag=""
-        if [[ -n "${KUBE_DOCKER_IMAGE_TAG-}" && -n "${KUBE_DOCKER_REGISTRY-}" && $docker_registry != $KUBE_DOCKER_REGISTRY ]]; then
-          release_docker_image_tag="${KUBE_DOCKER_REGISTRY}/${binary_name}-${arch}:${KUBE_DOCKER_IMAGE_TAG}"
+        if [[ -n "${KUBE_DOCKER_IMAGE_TAG-}" || -n "${KUBE_DOCKER_REGISTRY-}" ]]; then
+          release_docker_image_tag="${KUBE_DOCKER_REGISTRY-$docker_registry}/${binary_name}-${arch}:${KUBE_DOCKER_IMAGE_TAG-$docker_tag}"
```

Comment:
Instead of doing these inline substitutions, could set do something like:
```
KUBE_DOCKER_REGISTRY="${KUBE_DOCKER_REGISTRY-$docker_registry}"
```

That way, `$KUBE_DOCKER_REGISTRY` will always have a value (either an explicit one, or a sane default), rather than doing these inline ones. I haven't looked through the rest of the if logic here, but it's obvious we need to clean some of this up and make it more easily understandable.

Reply:
+1, this is pretty tricky to follow 😞

Reply:
what about now?
I always generate release_docker_image_tag, it could be default values or ENV value.
It should always remote the docker_tag image. Releases are going to be tagged with release_docker_image_tag, which should be different that docker_tag.